pkg/sqlite/database_postgres.go

```diff
@@ -0,0 +1,90 @@
+package sqlite
+
+import (
+	"fmt"
+
+	"github.com/doug-martin/goqu/v9"
+	_ "github.com/doug-martin/goqu/v9/dialect/postgres"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"github.com/jmoiron/sqlx"
+	"github.com/stashapp/stash/pkg/logger"
+)
+
+type PostgresDB Database
+
+func NewPostgresDatabase(dbConnector string) *Database {
+	dialect = goqu.Dialect("postgres")
+
+	db := &PostgresDB{
+		storeRepository: newDatabase(),
+		lockChan:        make(chan struct{}, 1),
+		dbConfig:        dbConnector,
+	}
+	db.dbInterface = db
+
+	dbWrapper.dbType = PostgresBackend
+
+	return (*Database)(db)
+}
+
+func (db *PostgresDB) DatabaseType() DatabaseType {
+	return PostgresBackend
+}
+
+func (db *PostgresDB) AppSchemaVersion() uint {
+	return uint(0 - (66 - int(appSchemaVersion)))
+}
+
+func (db *PostgresDB) DatabaseConnector() string {
+	return db.dbConfig.(string)
+}
+
+func (db *PostgresDB) open(disableForeignKeys bool, writable bool) (conn *sqlx.DB, err error) {
+	conn, err = sqlx.Open("pgx", db.DatabaseConnector())
+
+	if err != nil {
+		return nil, fmt.Errorf("db.Open(): %w", err)
+	}
+
+	if disableForeignKeys {
+		_, err = conn.Exec("SET session_replication_role = replica;")
+
+		if err != nil {
+			return nil, fmt.Errorf("conn.Exec(): %w", err)
+		}
+	}
+	if !writable {
+		_, err = conn.Exec("SET default_transaction_read_only = ON;")
+
+		if err != nil {
+			return nil, fmt.Errorf("conn.Exec(): %w", err)
+		}
+	}
+
+	return conn, nil
+}
+
+func (db *PostgresDB) Remove() error {
```

Comment:
I would suggest that we implement a set of methods like `CanRemove() bool`, `CanReset() bool` so then the Stash UI can hide the buttons accordingly

Reply:
I have not yet determined the solution for how to backup in postgresql.
Suggestions are welcome, current suggestions are:
- Fetch everything and like anonymise does and dump it.
- Force the user to create a backup file called "backup_postgres_{date}.gz", to attempt to force some backup.

Reply:
Well for now I think if the backend doesn't support it, just disabling the feature may make sense. That's why I was suggesting `CanRemove() bool`, `CanReset() bool`, etc... (or another option is to create an optional interface for `Remove() error` and then check if the object implements it at runtime)

As for backups, my .02 are that we shouldn't worry too much about backing up through Stash because we should let the DB admin do backups from the server.

The anonymization part is the most useful one. Maybe that could be done by creating a JSON dump?